backend/src/main/java/ca/gov/dtsstn/cdcp/api/web/v1/controller/UsersController.java

```diff
@@ -25,26 +35,74 @@
+		Assert.notNull(alertTypeService, "alertTypeService is required; it must not be null");
 		this.subscriptionModelAssembler = subscriptionModelAssembler;
 		this.subscriptionService = subscriptionService;
+		this.mapper = mapper;
+		this.alertTypeService = alertTypeService;
 	}
 
-	@GetMapping({ "/{id}/subscriptions" })
+	@GetMapping({ "/{userId}/subscriptions" })
 	@Operation(summary = "Get all subscriptions for a user.", operationId = "get-user-subscriptions")
 	public CollectionModel<SubscriptionModel> getSubscriptionsByUserId(
-			@NotBlank(message = "id must not be null or blank")
-			@Parameter(description = "The id of the user.", example = "000000000", required = true)
-			@PathVariable String id) {
-		final var subscriptions = subscriptionModelAssembler.toCollectionModel(subscriptionService.getSubscriptionsByUserId(id));
-		final var selfLink = linkTo(methodOn(getClass()).getSubscriptionsByUserId(id)).withSelfRel();
+			@NotBlank(message = "userId must not be null or blank")
+			@Parameter(description = "The user id of the user.", example = "000000000", required = true)
+			@PathVariable String userId) {
+		final var subscriptions = subscriptionModelAssembler.toCollectionModel(subscriptionService.getSubscriptionsByUserId(userId));
+		final var selfLink = linkTo(methodOn(getClass()).getSubscriptionsByUserId(userId)).withSelfRel();
 		return subscriptionModelAssembler.wrapCollection(subscriptions, SubscriptionModel.class).add(selfLink);
+	}	
+
+
+	@PostMapping({ "/{userId}/subscriptions/" })
+	@Operation(summary = "Create a subscription for a user.", operationId = "user-subscriptions-create")
+	@ApiResponse(responseCode = "204", description = "The request has been successfully processed.")
+	public void create(
+			@RequestBody(required = true)
+			SubscriptionModel createSubscriptionRequest) {
+		final Optional<AlertType> convertedAlertType = alertTypeService.readByCode(createSubscriptionRequest.getAlertType());
+		if (!convertedAlertType.isPresent()) {
+			log.warn("Can not retrieve a valid alertType from table: {}", createSubscriptionRequest.getAlertType());
+		}
+		SubscriptionModel createSubscriptionModel = createSubscriptionRequest;
+		createSubscriptionModel.setAlertType(convertedAlertType.get().getId());
+		final var subscription = mapper.toDomain(createSubscriptionModel);
+		log.debug("Creating subscription: {}", subscription);
+		subscriptionService.create(subscription);
 	}
 
+	@PostMapping({ "/{userId}/subscriptions/{subscriptionId}" })
+	@Operation(summary = "Update a subscription for a user.", operationId = "user-subscriptions-update")
+	@ApiResponse(responseCode = "204", description = "The request has been successfully processed.")
+	public void update(
+			@RequestBody(required = true)
+			SubscriptionModel updateSubscriptionRequest) {
+		final Optional<AlertType> convertedAlertType = alertTypeService.readByCode(updateSubscriptionRequest.getAlertType());
+		if (!convertedAlertType.isPresent()) {
```

Comment:
If the alert type is not found in the database, return a 400 status code.

Reply:
added